### PR TITLE
Add support for the InnerException in Json formatter using ToString()

### PR DESCRIPTION
**Pull request by maryamariyan**
Alternate solution for https://github.com/dotnet/runtime/issues/40507

### CODE REVIEW

**Comment by maryamariyan** [comment]
cc: @WernerMairl would you also review?

**Review by eerhardt — APPROVED** [review]
I think this is the best approach for `5.0`, which is fast closing. This matches the other 2 built-in formatters. And it also matches the behavior of the new Google APIs formatter as well:

https://github.com/googleapis/google-cloud-dotnet/blob/e7b75e6b40d7549d58aeb2dfe048ab271073d36f/apis/Google.Cloud.Logging.Console/Google.Cloud.Logging.Console/GoogleCloudConsoleFormatter.cs#L75

**Comment by maryamariyan** [comment]
@WernerMairl I favored your two tests over the one I wrote, by first [cherry-picking](https://github.com/dotnet/runtime/pull/40852/commits/04e504dcb5b043358be2091c47fdf8020adb107e) and [the third commit in this PR](https://github.com/dotnet/runtime/pull/40852/commits/6cb117fc08141e76553a63777e829a622f28412d) makes a little modification on top of the two tests two cover the assertions I was originally making.

File: src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/JsonConsoleFormatterTests.cs [hunk 1]
```diff
@@ -507,25 +478,46 @@ private string GetJson(Exception exception)
...
+            
+            Assert.Contains(GetContent(rootException, indented), json);
+            Assert.Contains(GetContent(rootException.InnerException, indented), json);
+        }
+
+        static string GetContent(Exception exception, bool indented)
+        {
+            string indentation = indented ? "\\r\\n" : " ";
+            return exception.ToString()
+                .Replace("\\", "\\\\")
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/JsonConsoleFormatterTests.cs:502]: needed for the stacktrace content that uses path `C:\\X\\Y\\Z` in the json for the path `C:\X\Y\Z`.

**Review by WernerMairl — APPROVED** [review]

**Comment by WernerMairl** [comment]
I agree 100% to do progress here by implementing pragmatic solutions :-)

But basically the entire "Exception.ToString()" thing is a nightmare regarding consistency and structured exception output management :-(

File: src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs [hunk 2]
```diff
@@ -105,6 +92,11 @@ public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeP
             textWriter.Write(Environment.NewLine);
         }
 
+        private static string WriteReplacing(string oldValue, string newValue, string message)
+        {
+            return message.Replace(oldValue, newValue);
```
Comment (danmosemsft) [inline src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs:97]: This does not seem like it is worth a method?

**Comment by WernerMairl** [comment]
My be the title of this PR is wrong !? 

"Including **Exception.Data** in Json formatter using ToString()"

using ToString() is solving "InnerException" issues.... 
"Exception.Data" is not covered by "ToString()" until now on master...

so please be careful on writing commit history ;-)

**Comment by maryamariyan** [comment]
```
if (!indented)
{
    result = result.Replace(Environment.NewLine, newLineReplacement);
}

return result;
```

The above wouldn't be sufficient because when not indented, new line gets replaced with " ".

Additionally, the purpose of having
```
.Replace(Environment.NewLine, newLineReplacement)
```

Was to have Environment.NewLine to get escaped (to "\\r\\n" or "\\n")

**Review by eerhardt — APPROVED** [review]
Looks good.

**Pull request merged**